### DISCUSSION TITLE
How to use affix state and mutable session togeather

### DISCUSSION THREAD

**Discussion by saumil-ventura1** [salvo-rs/salvo#723] [Q&A]
I am having problem using both mutable session from depot, and obtaining state from depot.
I am new to rust, so this might be silly.

here is e.g. code

```
#[handler]
pub async fn handler(depot: &mut Depot) {
    let session = depot.session_mut().unwrap();
    let db = depot.obtain::<DatabaseConnection>().unwrap();
    session.insert("", "");
   ...
}
```
When using above code, rust borrow checker gives error.
```
cannot borrow `*depot` as immutable because it is also borrowed as mutable
```
I need session to be mutable.

**Comment by chrislearn** [comment]
You can use `depot.take_session()` to take the session, if you need to use session in other handler, you can use `depot.set_session(session)` to put it into depot again.

This is not salvo's bug, just rust rule restriction.

**Reply by saumil-ventura1** [reply]
Ok, Thanks for help. This solved my issue.
